subworkflows/nf-core/bed_scatter_bedtools/main.nf

```diff
@@ -0,0 +1,45 @@
+include { BEDTOOLS_SPLIT     } from '../../../modules/nf-core/bedtools/split/main'
+
+workflow BED_SCATTER_BEDTOOLS {
+
+    // IMPORTANT: Add the configuration in nextflow.config file to your modules.config
+
+    take:
+    ch_bed          // channel: [ meta, bed, scatter_count ]
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    ch_bed
+        .map(
+            { meta, bed, scatter_count ->
+                meta = meta + [subwf_scatter_count:scatter_count]
```

Comment:
How about a more generic `scatter_count` key?

Reply:
My main thought was that I should create a key that isn't in the input meta. `scatter_count` might be a bit too common? I mean I'm not sure, but that's the reasoning behind it (I'll use the same value in the variant calling subwf)

Reply:
It's just that I'm not much of a fan of pushing extra meta keys on users 😅  get's real confusing real fast, especially if we intend to use the splitting function in conjunction with another subwf which isn't tailored to it.

Reply:
Yeah, I understand. Would it be better if I were to just add it as a constant value to each bed? the user can do with it whatever they want then.

Reply:
I think we should ask the #sarek overlords, they've got more experience with stuff like this.
@FriederikeHanssen , @maxulysse, opinions?

Reply:
@nvnieuwk not sure I understand what you were hinting at, can you elaborate?

Reply:
yes that works well. users can then put this in the meta map to use in other workflows: https://github.com/nf-core/sarek/blob/96749f742197e828f4632cc2a7481190e7f642ac/subworkflows/local/prepare_intervals/main.nf#L107

Reply:
Allrighty then! +1 on the value output

Reply:
allright, I'll fix it!

Reply:
Added!

Reply:
I guess this would be a general discussion point here. Should the scatter count be rather supplied as a param by the user inthe modules.config. Currently I guess usage ouwld be, pass in param in the pipeline -> add to bed file channel --> add to meta here to use in modules.config --> remove from meta again, add back to channel. Might be easier to just have the param directly accessed in the modules.config?

Reply:
I'm not really sure it should ever be added to meta in this workflow actually...

Reply:
Since the scatter count is a mandatory input, I'd treat it like a `val` input for the workflow.

Reply:
Yeah but the fact is that it could be dependent on the specific input file. That's the reason I put them together.